smap_l2_gridder/grid.py

```diff
@@ -29,23 +35,25 @@ def process_input(in_data: DataTree, output_file: Path, logger: None | Logger =
     """Process input file to generate gridded output file."""
     out_data = DataTree()
 
+    short_name = get_collection_shortname(in_data)
+
     out_data = transfer_metadata(in_data, out_data)
 
     # Process grids from all top level groups that are not only Metadata
-    data_node_names = set(in_data['/'].children) - set(get_metadata_children(in_data))
+    data_group_names = get_data_groups(in_data)
```

Comment:
I renamed the "nodes" to "groups" because that's more of what they actually are in terms of the file structure, it does add a few extra changes, but I think it makes the code better.